boards/shared/Inc/App/App_SharedSignal.h

```diff
@@ -41,26 +51,57 @@ struct Signal *App_SharedSignal_Create(
- * @param signal The signal to get the time from
- * @return The last time the given signal was observed to be low, in
- *         milliseconds
+ * Get the last time the given signal was observed to be low during entry, in
+ * milliseconds
+ * @param signal The signal to get the time during entry from
+ * @return The last time the given signal was observed to be low during entry,
+ * in milliseconds
+ */
+uint32_t App_SharedSignal_GetEntryLastTimeLowMs(const struct Signal *signal);
+
+/**
+ * Get the last time the given signal was observed to be high during entry, in
+ * milliseconds
+ * @param signal The signal to get the time during entry from
+ * @return The last time the given signal was observed to be high during entry,
+ * in milliseconds
+ */
+uint32_t App_SharedSignal_GetEntryLastTimeHighMs(const struct Signal *signal);
+
+/**
+ * Get the last time the given signal was observed to be low during exit, in
+ * milliseconds
+ * @param signal The signal to get the time during exit from
+ * @return The last time the given signal was observed to be low during exit, in
+ * milliseconds
+ */
+uint32_t App_SharedSignal_GetExitLastTimeLowMs(const struct Signal *signal);
+
+/**
+ * Get the last time the given signal was observed to be high during exit, in
+ * milliseconds
+ * @param signal The signal to get the time during exit from
+ * @return The last time the given signal was observed to be high during exit,
+ * in milliseconds
  */
-uint32_t App_SharedSignal_GetLastTimeLowMs(const struct Signal *signal);
+uint32_t App_SharedSignal_GetExitLastTimeHighMs(const struct Signal *signal);
 
 /**
- * Get the last time the given signal was observed to be high, in milliseconds
- * @param signal The signal to get the time from
- * @return The last time the given signal was observed to be high, in
- *         milliseconds
+ * Check if the callback function for the given signal is triggered
+ * @param signal The signal to check if the callback function is triggered
+ * @return true if the callback function is triggered, false if it is not
  */
-uint32_t App_SharedSignal_GetLastTimeHighMs(const struct Signal *signal);
+bool App_SharedSignal_IsCallbackTriggered(const struct Signal *const signal);
 
 /**
- * Update the internal state of the given signal. If the signal has been
- * continuously high for a period equal to or greater than the configured
- * duration (See: `high_duration_ms`), the callback function will be triggered.
- * @note This calls the is_high() method of the given signal
+ * Update the internal state of the given signal. If the entry condition for the
+ * signal has been continuously high for a period equal to or greater than the
+ * configured duration (See: `entry_high_duration_ms) the callback function will
+ * be triggered. The exit signal for the signal has to remain continuously high
+ * for a period equal to or greater than the configured duration (See:
+ * `exit_signal_duration_ms) to trigger the exit callback function.
```

Comment:
Since there actually is not exit callback function and entering the exit condition just means that you'll leave the entry callback function. Should this say something similar to:
The exit signal for the signal has to remain continuously high
 * for a period equal to or greater than the configured duration (See:
 * `exit_signal_duration_ms`) to exit the entry condition's callback function.

Reply:
Yeah this was a late night potato brain moment. This should be fixed